### PR TITLE
37 api authentication phase 2

### PR DESCRIPTION
**Pull request by malgo-ia**
Please verify and comment.

### CODE REVIEW

File: tests/utils/utils.py [hunk 1]
```diff
@@ -24,3 +24,6 @@ def uuid_generator() -> str:
         str
     """
     return str(uuid.uuid4())
+
+
+print(token_generator(16))
```
Comment (ginone) [inline tests/utils/utils.py:29]: to cleanup

File: app/test_draft/testy_put_login_manual.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import random
```
Comment (ginone) [inline app/test_draft/testy_put_login_manual.py:1]: to cleanup (whole file)

File: .github/workflows/pytest_and_docker_image.yml [hunk 3]
```diff
@@ -47,7 +47,7 @@ jobs:
     - name: install pytest-github-report
       run: pip install pytest-github-report
     - name: Run pytest unit
-      env:
+      env:
```
Comment (ginone) [inline .github/workflows/pytest_and_docker_image.yml:50]: to cleanup (whitespace)

File: app/endpoints/users_endpoints.py [hunk 4]
```diff
@@ -113,3 +114,28 @@ async def update_user_password(request: web.Request) -> web.Response:
...
+    except ValidationError as err:
+        return web.json_response({"error": str(err.messages)}, status=400)
+
+    with db.session() as db_session:
+        try:
+            token = users_services.verify_credentials(
+                db_session, body["login"], body["password"]
+            )
+        except Exception as e:
+            return web.json_response({"error": str(e)}, status=400)
```
Comment (ginone) [inline app/endpoints/users_endpoints.py:136]: it's a bad practice to return stack traces (`str(e)`) from database operations. we do not want hacker to learn about our database implementation (engine, version, table names, column names, etc)

please change errors to some generic messages in `try/except` blocks that perform db operations

File: app/services/users_services.py [hunk 5]
```diff
@@ -78,3 +79,29 @@ def update_user_password(db_session, user_id: int, new_password: str) -> None:
...
+    if not user:
+        raise Exception("login not existed in database")
+    password_hash = user.password
+
+    if not pbkdf2_sha256.verify(password, password_hash):
+        return False
+
+    token = jwt.encode(
+        {
+            "exp": datetime.utcnow() + timedelta(minutes=15),
```
Comment (ginone) [inline app/services/users_services.py:99]: eventually we will likely want to increase session to longer period, maybe even 24h?

**Review by ginone — CHANGES_REQUESTED** [review]

**Comment by malgo-ia** [comment]
Please check PR corrections. 
By mistake I also added changes related to  Issue #12 DB Session Middleware . Please let us close these two Issues together. Sorry for the inconvenience :)

File: app/services/users_services.py [hunk 6]
```diff
@@ -78,3 +79,29 @@ def update_user_password(db_session, user_id: int, new_password: str) -> None:
     users_queries.update_user_password(db_session, hashed_password, user_id)
 
     return None
+
+
+def verify_credentials(db_session, login: str, password: str) -> str:
```
Comment (ginone) [inline app/services/users_services.py:84]: IMO   "verify credentials" should  only verify credentials (basically return True or False),  creatingn JWT is separate resposibility, should be separate funnctionn

**Review by ginone — CHANGES_REQUESTED** [review]

**Review by ginone — APPROVED** [review]

**Pull request merged**